impeller/aiks/experimental_canvas.cc

```diff
@@ -371,13 +389,25 @@ void ExperimentalCanvas::SaveLayer(
       /*flood_input_coverage=*/!!backdrop_filter         //
   );
 
-  if (!maybe_subpass_coverage.has_value() ||
-      maybe_subpass_coverage->IsEmpty()) {
-    Save(total_content_depth);
-    return;
+  if (!maybe_subpass_coverage.has_value()) {
+    return SkipUntilMatchingRestore(total_content_depth);
   }
+
   auto subpass_coverage = maybe_subpass_coverage.value();
 
+  // When an image filter is present, clamp to avoid flicking due to nearest
+  // sampled image. For other cases, round out to ensure than any geometry is
+  // not cut off.
+  ISize subpass_size;
+  if (paint.image_filter) {
+    subpass_size = ISize(subpass_coverage.GetSize());
+  } else {
+    subpass_size = ISize(IRect::RoundOut(subpass_coverage).GetSize());
+  }
```

Comment:
Don't the variations of how you compute the subpass_size have a ramification for the origin? The subpass_coverage was relative to the global position I think? In which case if you just took its size and made a drawable for that then the global position would be the origin of the coverage. But if you rounded it out, then the global position would be the rounded down value of the origin of the coverage?

Reply:
hmm, good point. Right now we're always rounding in https://github.com/flutter/engine/blob/main/impeller/aiks/experimental_canvas.cc#L524-L526 

But I think instead we need to take into account whether we rounded in or out, so that we're not a pixel off.